Creating a _WKWebExtension with an invalid path should return nil.

xeenon opened the pull request:
#### c893edaa156c8a67f63eee27e62b3d8e283fb7a8
<pre>
Creating a _WKWebExtension with an invalid path should return nil.
<a href="https://bugs.webkit.org/show_bug.cgi?id=248960">https://bugs.webkit.org/show_bug.cgi?id=248960</a>

Reviewed by Brian Weinstein.

* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtension.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtension.mm:
(+[_WKWebExtension extensionWithAppExtensionBundle:]): Added.
(+[_WKWebExtension extensionWithResourceBaseURL:]): Added.
(-[_WKWebExtension initWithAppExtensionBundle:error:]): Added error param.
(-[_WKWebExtension initWithResourceBaseURL:error:]): Added error param.
(-[_WKWebExtension _initWithManifestDictionary:]): Added assert.
* Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionCocoa.mm:
(WebKit::WebExtension::WebExtension): Added error param.
* Source/WebKit/UIProcess/Extensions/WebExtension.h:
* Tools/TestWebKitAPI/cocoa/WebExtensionUtilities.mm:
(TestWebKitAPI::Util::loadAndRunExtension): Use initWithResourceBaseURL:error:.

Canonical link: <a href="https://commits.webkit.org/257584@main">https://commits.webkit.org/257584@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/99f0b9a12bbb0114dd667820d08b8a26dbfb138c

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/99366 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/8565 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/32487 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/108750 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/169000 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/103363 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/9127 "Built successfully") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/61/builds/85884 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/91855 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/106677 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/105124 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/6902 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/90441 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/33877 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/88721 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/21788 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/76753 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/81/builds/2435 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/23305 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/2339 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/61/builds/85884 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5221 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/8499 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/42780 "Passed tests") | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/79/builds/4206 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

b-weinstein commented on Source/WebKit/UIProcess/API/Cocoa/_WKWebExtension.mm:
```diff
@@ -42,34 +42,78 @@ @implementation _WKWebExtension
+    auto result = WebKit::WebExtension::create(appExtensionBundle, &internalError);
+
+    if (internalError)
+        return nil;
```
Do we have any way to log these? Do we want to?

b-weinstein replied:
Nit: This could be `return internalError ? nil : result->wrapper();`

xeenon replied:
We have `RELEASE_LOG(Extensions, "...")`.

b-weinstein commented on Source/WebKit/UIProcess/API/Cocoa/_WKWebExtension.mm:
```diff
@@ -42,34 +42,78 @@ @implementation _WKWebExtension
+    auto result = WebKit::WebExtension::create(resourceBaseURL, &internalError);
+
+    if (internalError)
+        return nil;
```
Ditto here. Or are we expecting clients to call the API with the error and perform the logging themselves?

xeenon replied:
I think the client should control logging or not for these.

xeenon committed:
Creating a _WKWebExtension with an invalid path should return nil.
https://bugs.webkit.org/show_bug.cgi?id=248960

Reviewed by Brian Weinstein.

* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtension.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtension.mm:
(+[_WKWebExtension extensionWithAppExtensionBundle:]): Added.
(+[_WKWebExtension extensionWithResourceBaseURL:]): Added.
(-[_WKWebExtension initWithAppExtensionBundle:error:]): Added error param.
(-[_WKWebExtension initWithResourceBaseURL:error:]): Added error param.
(-[_WKWebExtension _initWithManifestDictionary:]): Added assert.
* Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionCocoa.mm:
(WebKit::WebExtension::WebExtension): Added error param.
* Source/WebKit/UIProcess/Extensions/WebExtension.h:
* Tools/TestWebKitAPI/cocoa/WebExtensionUtilities.mm:
(TestWebKitAPI::Util::loadAndRunExtension): Use initWithResourceBaseURL:error:.

Canonical link: https://commits.webkit.org/257584@main
